doc/requirements/sdid.rst

```diff
@@ -0,0 +1,261 @@
+**************************************************
+Software Design and Implementation Document (SDID)
+**************************************************
+
+
+--------
+Overview
+--------
+
+..
+   TODO
+
+Settings Overview
+-----------------
+
+The settings package is a long-time feature of ARMI. It was made to fill the niche of gathering user input on how the user wanted a case to proceed and what results the user is hoping to produce.
+
+A good roadmap to understanding the settings system would be to look at the following principle design considerations, going into each one in detail.
+
+* Keep the user's experience with settings as simple as possible.
+* Remove setting definitions from code, centralize them, and facilitate definition change
+* Don't break backwards compatibility
+* Improve safety around possible misuses and accidents of the settings
+* Facilitate interactions with the GUI
+
+Outside of these points the system should exist in a form trimmed of any superfluous functionality to keep it clean and unobtrusive.
+
+Keep the user's experience with settings as simple as possible
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This is the principle design restraint for the settings package. In addition to raw implementation complexity, asking developers and users to relearn any alterations to the existing system counts towards total complexity; deeming migration to an equally complex system not worth the effort without stronger incentives.
+
+Historically developers are used to looking up the name of their setting by using a string fed into a collection object and having the value associated with the name returned back. The same behavior was used to assign a new value to the given name in the shared object. This constitutes the vast majority of external interactions with the settings system.
```

Comment:
The historical context feel unnecessary in this kind of document. Maybe it could be removed?